### ISSUE TITLE
Write up a technical glossary

### ISSUE THREAD

**Issue by Korosensei42** [diproche/webinterface#72]
In case other developers pick this project up in the future we should write up a glossary of all technical terms we use. This will also prevent us to mix two equal terms in their meaning or using them interchangibly.

What do you think.

**Comment by RonjaKaehne** [comment]
i dont think i got it clearly what you mean. technical terms like using the word "Spielwiese"? if you mean that yes we could add a wikipage inside this git repository.

**Comment by Korosensei42** [comment]
Yes, this could go in a wiki. I mean technical terms like "Spielwiese", "example", "textFormatter" and so on.
Documentation in the code is all fine and dandy, but in a glossary (essentially a dictionary with explenations) you would have all terms in one place.

**Comment by Korosensei42** [comment]
Stuff like grammar for certain Spielwiesen, as described in #46 should go in this glossary as well.

**Comment by Korosensei42** [comment]
From #46:

With regards to the Spielwiese Aussagenlogik the allowed grammar for formulae should be something like this:

A ::= a | b | ... | y | z (atoms)
F ::= \top | \bot | A | F \wedge F | F \vee F | \neg F | F \rightarrow F | F \leftrightarrow F (formulae)

Note: top and bot stand for verum and falsum respectively. The other keywords also correspond to their standard LaTeX-interpretations, i.e. they stand for and, or, negation, implication, equivalence.

How a sentence containing natural language or expressions from other Spielwiesen should look like, still needs to be done.

**Comment by TimothyGillespie** [comment]
I introduced / invented a term for the following case:

Assume we have a prolog knowledge basis as follows:

```
woman(anna). 
woman(eve). 
man(frank).
man(bob).
```

Then the query `?- woman(X).` will return `X = anna ; X = eve.`
In this case `anna` and `eve` are fill-in values for X in this query (the logic being prolog fills in possible values for the variable. I couldn't find an exisiting term for these.

**Comment by Korosensei42** [comment]
@TimothyGillespie Just so I understand you correctly: Do you have a term for this now or do you not? I don't find a term by you anywhere at least. Or do you mean the word `fill-in values`?

In the latter case: As known, `man` and `woman` are predicates. The arguments of predicates are always `terms`. `Terms` can also have so-called `sorts` (german: Sorten (oft) / Typen (selten) ). So you would say, that `anna` and `eve` are of sort `name` or `string`, depending on if you have defined an abstract data structure or type `name` or if you want to use the basic type `string`.

Also another term that would fit, would be: `anna` and `eve` are `termvaluations` over the Universe of sort `name` or `string` respectively. For the paragraph before and this one,

I will leave you a file down below of a little write-up by me. I can also include the previous explanation in the glossary, of course. I will start it on Saturday, I think.

In case you mean the kind of search that prolog does: I think `exhaustive search` fits it quite well, I think.

Here the image of the write-up:

[Termvaluation.pdf](https://github.com/diproche/webinterface/files/3360530/Termvaluation.pdf)

**Comment by TimothyGillespie** [comment]
I think `termvaluation` was the term I was looking for but couldn't find. Thanks!
